### ISSUE TITLE
Connecting alters resolution on Windows

### ISSUE THREAD

**Issue by Chealer** [TigerVNC/tigervnc#116]
We had an important issue when connection from TightVNC 2.7.10 on Windows 8 to a TightVNC 2.7.10 server on Windows 7. The resolution would change (for example from 1920x1080) to 1280x1024. Once connected, Windows would only let us choose between 800x600, 1024x768 and 1280x1024. We tried multiple settings without success.

We then switched to TigerVNC 1.4.2 on both sides, but the issue remains. This happens whether or not "Resize remote session on connect" is cheched, and whether or not "Resize remote session to the local window" is checked.

**Comment by Chealer** [comment]
This happens even if the screen is not connected when the Windows server boots (the screen probably has a maximal resolution of 1280x1024).

**Comment by CendioOssman** [comment]
Ehm. I don't see how this is possible. We don't have any code that fiddles with the screen resolution on Windows.

Are you sure you're connecting to our server? Is this a virtual machine? If so then you might be attaching to a VNC server that's part of the hypervisor.

**Comment by Chealer** [comment]
It is a physical machine. TigerVNC is the only VNC server installed currently.

**Comment by CendioOssman** [comment]
You write that this happens even without a monitor connected. But how do you know what resolution it is at beforehand without a monitor?

**Comment by Chealer** [comment]
The resolution in the RDP session is altered.

Note that the Windows installation of the server is not genuine.

**Comment by CendioOssman** [comment]
Huh? Is this a VNC server running in an RDP session rather than on the local console? I didn't even think that worked reliably.

**Comment by Chealer** [comment]
No, the server always runs (it runs as a service). If a session is already open, connecting to VNC will re-use that session after the login screen.

**Comment by CendioOssman** [comment]
Do you get the original resolutions back once you disconnect? Or do you need a reboot/logout?

**Comment by Chealer** [comment]
I am not sure if disconnecting properly speaking changes the resolution, but if I reconnect via RDP, the resolution is restored.

Note that it seems RDP connections are closed when one connects via VNC. I thought they stayed active. It may not be accurate to say that VNC alters the resolution then (the problem may be that it closes the RDP session).
When a user connects via RDP, any VNC session of the user is closed.

**Comment by CendioOssman** [comment]
Ah. That might be a limitation of how things work I'm afraid. Connecting via RDP generally disconnects you from the local console. And our way of attaching to the desktop might no longer work in that case. Correspondingly, we might try to make the session active on the local console when someone connects, which probably forces out the RDP connection.
